File: The Braithwaite Confront/sol.java
import java.util.Scanner;

public class Main {

    public static int max(int a, int b) {
        return (a >= b) ? a : b;
    }

    public static void solve() {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        int r = scanner.nextInt();
        int c = scanner.nextInt();
        System.out.println(max(n - r, r - 1) + max(m - c, c - 1));
        scanner.close();
    }

    public static void main(String[] args) {
        solve();
    }
}

File: The Braithwaite Confront/sol.cpp
#include <iostream>
using namespace std;

int max(int a, int b) {
    return (a >= b) ? a : b;
}

void solve() {
    int n, m, r, c;
    cin >> n >> m >> r >> c;
    cout << max(n - r, r - 1) + max(m - c, c - 1) << endl;
}

int main() {
    solve();
    return 0;
}
